sql/analyzer/hoist_select_exists.go

```diff
@@ -260,6 +260,18 @@ func decorrelateOuterCols(sqChild sql.Node, aliasDisambig *aliasDisambiguator, c
 				if gf, ok := e.(*expression.GetField); ok && corr.Contains(gf.Id()) {
 					return true
 				}
+				if sq, ok := e.(*plan.Subquery); ok {
```

Comment:
subqueries have a list of correlated columns inside, can we intersect those with corr?